### PR TITLE
ML: Add useRemoveFolder hook

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

This PR adds a new hook called `userRemoveFolder` which will be used to delete a folder. It also adds tests for `useRemoveAsset`.

### Why is it needed?

To be able to delete folders from the media library.

### How to test it?

Rely on the automated tests for now.

### CODE REVIEW

File: packages/core/upload/admin/src/hooks/useRemoveFolder.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { useNotification } from '@strapi/helper-plugin';
+import { deleteRequest } from '../utils/deleteRequest';
+
+export const useRemoveFolder = () => {
+  const toggleNotification = useNotification();
+  const queryClient = useQueryClient();
+
+  const mutation = useMutation(id => deleteRequest('folders', id), {
+    onSuccess: () => {
+      queryClient.refetchQueries(['folders'], { active: true });
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useRemoveFolder.js:11]: Can you add rename the query to `[pluginId, 'folders']` ? So we know that there would only be one folders query in the entire project.

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/upload/admin/src/hooks/useRemoveFolder.js (hunk 1 above)
Comment (gu-stav) [reply packages/core/upload/admin/src/hooks/useRemoveFolder.js:11]: Do you think we should do this for all `upload` related queries then? Meaning: `assets`, `assets-count`?

**Comment by gu-stav** [comment]
@soupette This one should be ready for review 🚀

File: packages/core/upload/admin/src/hooks/tests/useRemoveAsset.test.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import { NotificationsProvider, useNotification } from '@strapi/helper-plugin';
+import { deleteRequest } from '../../utils/deleteRequest';
+
+import { useRemoveAsset } from '../useRemoveAsset';
+
+const ASSET_FIXTURE = {
+  id: 1,
+};
+
+console.error = jest.fn();
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/tests/useRemoveAsset.test.js:15]: What kind of error do we have in the console that requires to mock the console?

File: packages/core/upload/admin/src/hooks/useRemoveAsset.js [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
   const toggleNotification = useNotification();
   const queryClient = useQueryClient();
 
-  const mutation = useMutation(assetId => removeAssetRequest(assetId), {
+  const mutation = useMutation(assetId => deleteRequest('files', assetId), {
     onSuccess: () => {
-      // Coupled with the cache of useAssets
-      queryClient.refetchQueries(['assets'], { active: true });
-      queryClient.refetchQueries(['asset-count'], { active: true });
+      queryClient.refetchQueries([pluginId, 'assets'], { active: true });
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useRemoveAsset.js:13]: What's the difference between refetchQueries and invalidateQuery ?

**Review by soupette — APPROVED** [review]
I just have 2 comments but we can merge it as it is!

Good job!

**Pull request merged**